File: Harish Kumar/Dynamic Programming & Recursion/ Count Sorted Vowel Strings/ Count Sorted Vowel Strings.java
class Solution {
    public int countVowelStrings(int n) {
        
        int i = 0;
        int[] dp = new int[5];
        
        while (i < 5) {
            dp[i] = ++i;
        }
        
        while (--n != 0) {
            for (i = 1; i < 5; i++) {
                dp[i] += dp[i - 1];
            }
        }
        
        return dp[4];
    }
}

File: Aman/DP Recursion and Backtracking/Count Sorted Vowel Strings.cpp
class Solution {
public:
    int countVowelStrings(int n) {
        
        vector<int>dp(5,1);
        int res=0;
        for(int i=0;i<n-1;i++)
        {
            for(int j=3;j>=0;j--)
            {
                dp[j]+=dp[j+1];
            }
        }
        
        for(int i=0;i<5;i++)
            res+=dp[i];
        
        return res;
    }
};
